ml4h/models.py

```diff
@@ -1273,6 +1722,157 @@ def make_paired_autoencoder_model(
     return m, encoders, decoders
 
 
+BLOCK_CLASSES = {
+    'conv_encode': ConvEncoderBlock,
+    'conv_decode': ConvDecoderBlock,
+    'concat': FlatConcatDenseBlock,
+    'average': AverageBlock,
+    'flat': FlatDenseBlock,
+    'pair': PairLossBlock,
+    'gap': GlobalAveragePoolBlock,
+    'lstm_encode': LSTMEncoderBlock,
+    'language_decode': LanguageDecoderBlock,
+    'fc': FullyConnectedBlockBlock,
+    'fc_decode': DenseDecoderBlock,
+    'identity': EncodeIdentityBlock,
+}
+
+
+def compose(f, g):
+    return lambda x, y: g(f(x, y), y)
+
+
+def identity(x, _):
+    return x
+
+
+def block_make_multimodal_multitask_model(
+        tensor_maps_in: List[TensorMap],
+        tensor_maps_out: List[TensorMap],
+        encoder_blocks: List[str],
+        decoder_blocks: List[str],
+        merge_blocks: List[str],
+        learning_rate: float,
+        optimizer: str,
+        u_connect: DefaultDict[TensorMap, Set[TensorMap]] = None,
+        training_steps: int = None,
+        learning_rate_schedule: str = None,
+        **kwargs,
+) -> Tuple[Model, Dict[TensorMap, Model], Dict[TensorMap, Model]]:
+    """Make multi-task, multi-modal feed forward neural network for all kinds of prediction
+
+    This model factory can be used to make networks for classification, regression, and segmentation
+    The tasks attempted are given by the output TensorMaps.
+    The modalities and the first layers in the architecture are determined by the input TensorMaps.
+    Model architectures are specified by Blocks which can encode, merge or decode TensorMaps.
+
+    Hyperparameters are exposed to the command line and passed through to Block constructors via **kwargs.
+    Model summary printed to output
+
+    :param tensor_maps_in: List of input TensorMaps
+    :param tensor_maps_out: List of output TensorMaps
+    :param encoder_blocks: One or more BLOCK_CLASS keys or hd5 model files to encode applicable input TensorMaps
```

Comment:
Could this instead be a dictionary from tensormap to list of blocks to apply to that tensormap?
E.g. 
```
{
  ecg_tmap: [denseblock(32 channels), denseblock(64 channels)]
  age_tmap: [fullyconnected_block(32 units)]
}
```